osgp/platform/osgp-adapter-ws-smartmetering/src/test/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/ActualPowerQualityResponseMappingTest.java

```diff
@@ -0,0 +1,137 @@
+        // Arrange
+        final List<PowerQualityObject> powerQualityObjects = this.makePowerQualityObjects();
+        final List<PowerQualityValue> powerQualityValues = this.makePowerQualityValues();
+
+        final ActualPowerQualityData responseData = new ActualPowerQualityData(powerQualityObjects,
+                powerQualityValues);
+
+        final ActualPowerQualityResponse source = new ActualPowerQualityResponse(responseData);
+
+        // Act
+        final org.opensmartgridplatform.adapter.ws.schema.smartmetering.bundle.ActualPowerQualityResponse target
+                = this.monitoringMapper
+                .map(source,
+                        org.opensmartgridplatform.adapter.ws.schema.smartmetering.bundle.ActualPowerQualityResponse.class);
+
+        // Assert
+        final org.opensmartgridplatform.adapter.ws.schema.smartmetering.monitoring.ActualPowerQualityData mappedResponseData = target
+                .getActualPowerQualityData();
+
+        this.assertPowerQualityObjects(mappedResponseData.getPowerQualityObjects().getPowerQualityObject(),
+                powerQualityObjects);
+        this.assertPowerQualityValues(mappedResponseData.getPowerQualityValues().getPowerQualityValue(), powerQualityValues);
+
+    }
+
+    @Test
+    public void sourceNull() {
+        final org.opensmartgridplatform.adapter.ws.schema.smartmetering.bundle.ActualPowerQualityResponse target
+                = this.monitoringMapper
+                .map(null,
+                        org.opensmartgridplatform.adapter.ws.schema.smartmetering.bundle.ActualPowerQualityResponse.class);
+
+        assertThat(target).isNull();
+    }
+
+    private void assertPowerQualityObject(
+            final org.opensmartgridplatform.adapter.ws.schema.smartmetering.monitoring.PowerQualityObject actualPowerQualityObject,
+            final PowerQualityObject sourcePowerQualityObject) throws AssertionError {
+
+        assertThat(actualPowerQualityObject.getName()).as(MAPPED_FIELD_VALUE_MESSAGE)
+                                                        .isEqualTo(sourcePowerQualityObject.getName());
+        assertThat(actualPowerQualityObject.getUnit().value()).as(MAPPED_FIELD_VALUE_MESSAGE)
+                                                      .isEqualTo(sourcePowerQualityObject.getUnit());
+    }
+
+    private void assertPowerQualityObjects(
+            final List<org.opensmartgridplatform.adapter.ws.schema.smartmetering.monitoring.PowerQualityObject> actualPowerQualityObjects,
+            final List<PowerQualityObject> sourcePowerQualityObjects) throws AssertionError {
+
+        assertThat(actualPowerQualityObjects.size()).as(MAPPED_LIST_SIZE_MESSAGE).isEqualTo(sourcePowerQualityObjects.size());
+
+        IntStream.range(0, actualPowerQualityObjects.size())
+                .forEach(i -> this.assertPowerQualityObject(actualPowerQualityObjects.get(i),
+                        sourcePowerQualityObjects.get(i)));
```

Comment:
implemented

Reply:
Line 104 in `assertPowerQualityValues` still streams to get an iteration index.

Reply:
If you change the other iteration, it would be good if you could make sure formatting in this new class is applied consistently according to the conventions.

Reply:
You're right. I had a complete focus on the lines mentioned. There is only one other usage of the IntStream class in GXF in the FindEventsCommandExecutorTest class. Could this usage have you're approval?

Reply:
I think you refer to the code that starts like `IntStream.rangeClosed(77, 89).forEach(code -> {`.
While you could debate whether a for loop would be better or not there, I think it is a subtly different case, where the stream looks less far-fetched.
Using a for loop with an index i from 0 to collection length is more of a standard idiom than iterating over a range of codes, which are used where the code has a specific meaning other than an array index.